module/VuFind/src/VuFind/Recommend/Databases.php

```diff
@@ -272,7 +272,10 @@ public function getResults()
 
         // Add databases from search query
         if ($this->useQuery) {
-            $query = strtolower($this->results->getParams()->getQuery()->getString());
+            $queryObject = $this->results->getParams()->getQuery();
+            $query = method_exists($queryObject, 'getString')
+                ? strtolower($queryObject->getString())
+                : null;
```

Comment:
I think you should default to `''` instead of `null` since it is illegal to call `strlen()` on `null`.